test.py

```diff
@@ -36,6 +36,7 @@
 from test.pylib.artifact_registry import ArtifactRegistry
 from test.pylib.host_registry import HostRegistry
 from test.pylib.pool import Pool
+from test.pylib.s3_proxy import S3ProxyServer
```

Comment:
the commit message reads:

> test: Switch s3_test to use proxy
>
> Switch `s3_test` to use the S3 proxy, fix `put_object` to make it retryable

but it does not explain why we need to have a proxy sitting in-between the client and minio. presumably, it for error injection. could you please elaborate it a little bit in the commit message?

Reply:
Done